### PR TITLE
[do not merge] nubis user management

### PR DESCRIPTION
**Pull request by limed**
This manages users by querying ldap and inserting into consul, this is not the IAM portion yet

### CODE REVIEW

**Comment by limed** [comment]
This PR is a bit messy and if you would like I can try to squash them down

File: modules/vpc/main.tf [hunk 1]
```diff
@@ -1340,3 +1340,163 @@ resource "aws_s3_bucket_object" "public_state" {
...
+                "ec2:DescribeNetworkInterfaces",
+                "ec2:DeleteNetworkInterface"
+            ],
+            "Resource": "*"
+        },
+        {
+            "Sid": "CreateIAMUsers",
+            "Effect": "Allow",
+            "Action": [
+                "iam:List*",
```
Comment (gozer) [inline modules/vpc/main.tf]: I am sure opsec would want us to be more specific here

File: modules/vpc/main.tf [hunk 2]
```diff
@@ -1340,3 +1340,163 @@ resource "aws_s3_bucket_object" "public_state" {
...
+
+    lifecycle {
+        create_before_destroy = true
+    }
+
+    vpc_id      = "${element(aws_vpc.nubis.*.id, count.index)}"
+    name_prefix = "MocoLdapOutbound-${element(split(",", var.environments), count.index)}-"
+    description = "Allow outbound ldap connection to moco ldap"
+
+    egress {
```
Comment (gozer) [inline modules/vpc/main.tf]: First, is this egress rule even strictly speaking necessary?
Second, you are apparently hard-coding a Mozilla IP address in there, how could we avoid that?

File: modules/vpc/main.tf [hunk 3]
```diff
@@ -1340,3 +1340,163 @@ resource "aws_s3_bucket_object" "public_state" {
...
+        TechnicalContact    = "${var.technical_contact}"
+        Environment         = "${element(split(",",var.environments), count.index)}"
+    }
+}
+
+resource "aws_cloudwatch_event_rule" "user_management_event_consul" {
+    count               = "${var.enabled * var.enable_user_management * length(split(",", var.environments))}"
+    name                = "user_management-consul-${element(split(",", var.environments), count.index)}"
+    description         = "Sends payload over a periodic time"
+    schedule_expression = "rate(15 minutes)"
```
Comment (gozer) [inline modules/vpc/main.tf]: I would probably expose this one as a variable, defaulting to rate(15 minutes), but not entirely sure how likely an account would ever want a different value

File: modules/vpc/main.tf [hunk 4]
```diff
@@ -1340,3 +1340,163 @@ resource "aws_s3_bucket_object" "public_state" {
...
+{
+    "command": "./nubis-user-management",
+    "args": [
+        "-execType=consul",
+        "-useDynamo=true",
+        "-region=${var.aws_region}",
+        "-environment='${element(split(",", var.environments), count.index)}'",
+        "-service=nubis",
+        "-accountName=${var.account_name}",
+        "-consulDomain=nubis.allizom.org",
```
Comment (gozer) [inline modules/vpc/main.tf]: nubis_domain is a variable somewhere, let's try and not hardcode .allizom.org anywhere if we can help it

File: modules/vpc/main.tf [hunk 5]
```diff
@@ -1340,3 +1340,163 @@ resource "aws_s3_bucket_object" "public_state" {
...
+    schedule_expression = "rate(15 minutes)"
+}
+
+resource "aws_cloudwatch_event_target" "user_management_consul" {
+    count       = "${var.enabled * var.enable_user_management * length(split(",", var.environments))}"
+    rule        = "user_management-consul-${element(split(",", var.environments), count.index)}"
+    arn         = "${element(aws_lambda_function.user_management.*.arn, count.index)}"
+    input       = <<EOF
+{
+    "command": "./nubis-user-management",
```
Comment (gozer) [inline modules/vpc/main.tf]: I like this kind of abstraction/decoupling, nice!

**Comment by limed** [comment]
I'm closing this PR, this PR is way to messy and I'm going to submit another one thats a little cleaner

**Pull request closed without merging**